### ISSUE TITLE
Failed building wheel for groundingdino

### ISSUE THREAD

**Issue by guytakeiteasy** [IDEA-Research/GroundingDINO#78]
Installing sd-webui-segment-anything requirement: groundingdino
提示：SD-WebUI 正在联网同步内部组件，这可能需要几秒到几十分钟不等，请耐心等待，切勿中途退出，以免文件损坏。切换版本后在此挂起一段时间是正常情况，如果您在使用中反复需要在此处等待 5 分钟以上，这可能是由于您安装的扩展与主环境产生版本冲突。请尝试通过扩展管理工具手动对齐扩展版本，或停用可疑扩展。
详见: https://support.qq.com/products/488228/faqs/135024
Traceback (most recent call last):
  File "D:\tools\Stable diffusion\SD-webui-aki-v3\extensions\sd-webui-segment-anything\scripts\dino.py", line 36, in install_goundingdino
    launch.run_pip(
  File "D:\tools\Stable diffusion\SD-webui-aki-v3\launch.py", line 129, in run_pip
    return run(f'"{python}" -m pip {args} --prefer-binary{index_url_line}', desc=f"Installing {desc}", errdesc=f"Couldn't install {desc}")
  File "D:\tools\Stable diffusion\SD-webui-aki-v3\launch.py", line 97, in run
    raise RuntimeError(message)
RuntimeError: Couldn't install sd-webui-segment-anything requirement: groundingdino.
提示：Python 运行时抛出了一个异常。请检查疑难解答页面。
Command: "D:\tools\Stable diffusion\SD-webui-aki-v3\py310\python.exe" -m pip install git+https://github.com/IDEA-Research/GroundingDINO --prefer-binary --index-url https://mirrors.bfsu.edu.cn/pypi/web/simple
Error code: 1
stdout: Looking in indexes: https://mirrors.bfsu.edu.cn/pypi/web/simple, https://pypi.tuna.tsinghua.edu.cn/simple, https://pypi.doubanio.com/simple

Looking in links: https://mirror.sjtu.edu.cn/pytorch-wheels/torch_stable.html

Collecting git+https://github.com/IDEA-Research/GroundingDINO

  Cloning https://github.com/IDEA-Research/GroundingDINO to c:\users\jack\appdata\local\temp\pip-req-build-de7pe97k

  Resolved https://github.com/IDEA-Research/GroundingDINO to commit 498048b1b2b770aca3f98430167dbbf458915c18

  Preparing metadata (setup.py): started

  Preparing metadata (setup.py): finished with status 'done'

Requirement already satisfied: torch in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (2.0.0+cu118)

Requirement already satisfied: torchvision in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (0.15.1+cu118)

Requirement already satisfied: transformers in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (4.25.1)

Requirement already satisfied: addict in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (2.4.0)

Requirement already satisfied: yapf in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (0.32.0)

Requirement already satisfied: timm in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (0.6.7)

Requirement already satisfied: numpy in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (1.23.3)

Requirement already satisfied: opencv-python in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from groundingdino==0.1.0) (4.6.0.66)

Collecting supervision==0.6.0

  Using cached https://pypi.doubanio.com/packages/42/7a/4fe657cae29eedeca34756a80a053ac3b1b3c80462cdb6768f428625f9ca/supervision-0.6.0-py3-none-any.whl (31 kB)

Collecting pycocotools

  Using cached https://pypi.doubanio.com/packages/ef/c6/90220be3b39fbc4cbd203775ca47dd8dc97fae06fbd2b500637395621b7c/pycocotools-2.0.6.tar.gz (24 kB)

  Installing build dependencies: started

  Installing build dependencies: finished with status 'done'

  Getting requirements to build wheel: started

  Getting requirements to build wheel: finished with status 'done'

  Preparing metadata (pyproject.toml): started

  Preparing metadata (pyproject.toml): finished with status 'done'

Requirement already satisfied: matplotlib in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from supervision==0.6.0->groundingdino==0.1.0) (3.6.2)

Requirement already satisfied: networkx in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torch->groundingdino==0.1.0) (2.8.8)

Requirement already satisfied: filelock in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torch->groundingdino==0.1.0) (3.8.0)

Requirement already satisfied: jinja2 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torch->groundingdino==0.1.0) (3.1.2)

Requirement already satisfied: sympy in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torch->groundingdino==0.1.0) (1.11.1)

Requirement already satisfied: typing-extensions in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torch->groundingdino==0.1.0) (4.4.0)

Requirement already satisfied: requests in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torchvision->groundingdino==0.1.0) (2.25.1)

Requirement already satisfied: pillow!=8.3.*,>=5.3.0 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from torchvision->groundingdino==0.1.0) (9.4.0)

Requirement already satisfied: packaging>=20.0 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (21.3)

Requirement already satisfied: pyyaml>=5.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (6.0)

Requirement already satisfied: huggingface-hub<1.0,>=0.10.0 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (0.13.4)

Requirement already satisfied: tqdm>=4.27 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (4.64.1)

Requirement already satisfied: tokenizers!=0.11.3,<0.14,>=0.11.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (0.12.1)

Requirement already satisfied: regex!=2019.12.17 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from transformers->groundingdino==0.1.0) (2022.10.31)

Requirement already satisfied: contourpy>=1.0.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (1.0.6)

Requirement already satisfied: pyparsing>=2.2.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (3.0.9)

Requirement already satisfied: fonttools>=4.22.0 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (4.38.0)

Requirement already satisfied: python-dateutil>=2.7 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (2.8.2)

Requirement already satisfied: cycler>=0.10 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (0.11.0)

Requirement already satisfied: kiwisolver>=1.0.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from matplotlib->supervision==0.6.0->groundingdino==0.1.0) (1.4.4)

Requirement already satisfied: colorama in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from tqdm>=4.27->transformers->groundingdino==0.1.0) (0.4.6)

Requirement already satisfied: MarkupSafe>=2.0 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from jinja2->torch->groundingdino==0.1.0) (2.1.1)

Requirement already satisfied: urllib3<1.27,>=1.21.1 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from requests->torchvision->groundingdino==0.1.0) (1.26.12)

Requirement already satisfied: idna<3,>=2.5 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from requests->torchvision->groundingdino==0.1.0) (2.10)

Requirement already satisfied: certifi>=2017.4.17 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from requests->torchvision->groundingdino==0.1.0) (2022.9.24)

Requirement already satisfied: chardet<5,>=3.0.2 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from requests->torchvision->groundingdino==0.1.0) (4.0.0)

Requirement already satisfied: mpmath>=0.19 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from sympy->torch->groundingdino==0.1.0) (1.3.0)

Requirement already satisfied: six>=1.5 in d:\tools\stable diffusion\sd-webui-aki-v3\py310\lib\site-packages (from python-dateutil>=2.7->matplotlib->supervision==0.6.0->groundingdino==0.1.0) (1.16.0)

Building wheels for collected packages: groundingdino, pycocotools

  Building wheel for groundingdino (setup.py): started

  Building wheel for groundingdino (setup.py): finished with status 'error'

  Running setup.py clean for groundingdino

  Building wheel for pycocotools (pyproject.toml): started

  Building wheel for pycocotools (pyproject.toml): finished with status 'error'

Failed to build groundingdino pycocotools

stderr:   Running command git clone --filter=blob:none --quiet https://github.com/IDEA-Research/GroundingDINO 'C:\Users\jack\AppData\Local\Temp\pip-req-build-de7pe97k'

  error: subprocess-exited-with-error

  × python setup.py bdist_wheel did not run successfully.

  │ exit code: 1

  ╰─> [45 lines of output]

      Building wheel groundingdino-0.1.0

      Compiling with CUDA

      running bdist_wheel

      running build

      running build_py

      creating build

      creating build\lib.win-amd64-cpython-310

      creating build\lib.win-amd64-cpython-310\groundingdino

      copying groundingdino\version.py -> build\lib.win-amd64-cpython-310\groundingdino

      copying groundingdino\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino

      creating build\lib.win-amd64-cpython-310\groundingdino\datasets

      copying groundingdino\datasets\transforms.py -> build\lib.win-amd64-cpython-310\groundingdino\datasets

      copying groundingdino\datasets\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino\datasets

      creating build\lib.win-amd64-cpython-310\groundingdino\models

      copying groundingdino\models\registry.py -> build\lib.win-amd64-cpython-310\groundingdino\models

      copying groundingdino\models\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino\models

      creating build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\box_ops.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\get_tokenlizer.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\inference.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\logger.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\misc.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\slconfig.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\slio.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\time_counter.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\utils.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\visualizer.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\vl_utils.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      copying groundingdino\util\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino\util

      creating build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\bertwarper.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\fuse_modules.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\groundingdino.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\ms_deform_attn.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\transformer.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\transformer_vanilla.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\utils.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      copying groundingdino\models\GroundingDINO\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO

      creating build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO\backbone

      copying groundingdino\models\GroundingDINO\backbone\backbone.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO\backbone

      copying groundingdino\models\GroundingDINO\backbone\position_encoding.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO\backbone

      copying groundingdino\models\GroundingDINO\backbone\swin_transformer.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO\backbone

      copying groundingdino\models\GroundingDINO\backbone\__init__.py -> build\lib.win-amd64-cpython-310\groundingdino\models\GroundingDINO\backbone

      running build_ext

      error: [WinError 2] 系统找不到指定的文件。

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.

  ERROR: Failed building wheel for groundingdino

  error: subprocess-exited-with-error

  × Building wheel for pycocotools (pyproject.toml) did not run successfully.

  │ exit code: 1

  ╰─> [23 lines of output]

      running bdist_wheel

      running build

      running build_py

      creating build

      creating build\lib.win-amd64-cpython-310

      creating build\lib.win-amd64-cpython-310\pycocotools

      copying pycocotools\coco.py -> build\lib.win-amd64-cpython-310\pycocotools

      copying pycocotools\cocoeval.py -> build\lib.win-amd64-cpython-310\pycocotools

      copying pycocotools\mask.py -> build\lib.win-amd64-cpython-310\pycocotools

      copying pycocotools\__init__.py -> build\lib.win-amd64-cpython-310\pycocotools

      running build_ext

      building 'pycocotools._mask' extension

      creating build\temp.win-amd64-cpython-310

      creating build\temp.win-amd64-cpython-310\Release

      creating build\temp.win-amd64-cpython-310\Release\common

      creating build\temp.win-amd64-cpython-310\Release\pycocotools

      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\lib\site-packages\numpy\core\include" -I./common "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\include" "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\Include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tc./common/maskApi.c /Fobuild\temp.win-amd64-cpython-310\Release\./common/maskApi.obj
None

      maskApi.c

GroundingDINO install failed. Please submit an issue to https://github.com/IDEA-Research/Grounded-Segment-Anything/issues.
      ./common/maskApi.c(151): warning C4101: \xa1\xb0xp\xa1\xb1: δ\xd2\xfd\xd3õľֲ\xbf\xb1\xe4\xc1\xbf

      "C:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\bin\HostX86\x64\cl.exe" /c /nologo /O2 /W3 /GL /DNDEBUG /MD "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\lib\site-packages\numpy\core\include" -I./common "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\include" "-ID:\tools\Stable diffusion\SD-webui-aki-v3\py310\Include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Tools\MSVC\14.35.32215\ATLMFC\include" "-IC:\Program Files (x86)\Microsoft Visual Studio\2022\BuildTools\VC\Auxiliary\VS\include" "-IC:\Program Files (x86)\Windows Kits\10\include\10.0.22000.0\ucrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\um" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\shared" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\winrt" "-IC:\Program Files (x86)\Windows Kits\10\\include\10.0.22000.0\\cppwinrt" "-IC:\Program Files (x86)\Windows Kits\NETFXSDK\4.8\include\um" /Tcpycocotools/_mask.c /Fobuild\temp.win-amd64-cpython-310\Release\pycocotools/_mask.obj

      _mask.c

      c1: fatal error C1083: \xce޷\xa8\xb4\xf2\xbf\xaaԴ\xceļ\xfe: \xa1\xb0pycocotools/_mask.c\xa1\xb1: No such file or directory

      error: command 'C:\\Program Files (x86)\\Microsoft Visual Studio\\2022\\BuildTools\\VC\\Tools\\MSVC\\14.35.32215\\bin\\HostX86\\x64\\cl.exe' failed with exit code 2

      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.

  ERROR: Failed building wheel for pycocotools

ERROR: Could not build wheels for pycocotools, which is required to install pyproject.toml-based projects

env: python:3.10.8 CUDA:11.8 Torch:2.0.0
what's the problem? can someone help me?

**Comment by HowieHepburn** [comment]
have you solve the problem? "ERROR: Failed building wheel for pycocotools"

**Comment by guytakeiteasy** [comment]
not yet:-(

**Comment by HowieHepburn** [comment]
邮件已收到

**Comment by HowieHepburn** [comment]
I've solved this problem by installing Microsoft C++ build tools, but when I read your error message carefully, I realized that it was different from yours.

**Comment by guytakeiteasy** [comment]
> I've solved this problem by installing Microsoft C++ build tools, but when I read your error message carefully, I realized that it was different from yours.

i try to install VC++2017 again,then i get a new error as below:

Start SAM Processing
Installing sd-webui-segment-anything requirement: groundingdino
提示：SD-WebUI 正在联网同步内部组件，这可能需要几秒到几十分钟不等，请耐心等待，切勿中途退出，以免文件损坏。切换版本后在此挂起一段时间是正常情况，如果您在使用中反复需要在此处等待 5 分钟以上，这可能是由于您安装的扩展与主环境产生版本冲突。请尝试通过扩展管理工具手动对齐扩展版本，或停用可疑扩展。
详见: https://support.qq.com/products/488228/faqs/135024
Traceback (most recent call last):
  File "L:\迅雷下载\Stable diffusion(3)\SD-webui-aki-v3\extensions\sd-webui-segment-anything\scripts\dino.py", line 36, in install_goundingdino
    launch.run_pip(
  File "L:\迅雷下载\Stable diffusion(3)\SD-webui-aki-v3\launch.py", line 129, in run_pip
    return run(f'"{python}" -m pip {args} --prefer-binary{index_url_line}', desc=f"Installing {desc}", errdesc=f"Couldn't install {desc}")
  File "L:\迅雷下载\Stable diffusion(3)\SD-webui-aki-v3\launch.py", line 97, in run
    raise RuntimeError(message)
RuntimeError: Couldn't install sd-webui-segment-anything requirement: groundingdino.
提示：Python 运行时抛出了一个异常。请检查疑难解答页面。
Command: "L:\迅雷下载\Stable diffusion(3)\SD-webui-aki-v3\py310\python.exe" -m pip install git+https://github.com/IDEA-Research/GroundingDINO --prefer-binary --index-url https://mirrors.bfsu.edu.cn/pypi/web/simple
Error code: 1
stdout: Looking in indexes: https://mirrors.bfsu.edu.cn/pypi/web/simple, https://pypi.tuna.tsinghua.edu.cn/simple, https://pypi.doubanio.com/simple

Looking in links: https://mirror.sjtu.edu.cn/pytorch-wheels/torch_stable.html

Collecting git+https://github.com/IDEA-Research/GroundingDINO

  Cloning https://github.com/IDEA-Research/GroundingDINO to c:\users\zx\appdata\local\temp\pip-req-build-3ase2mik

stderr:   Running command git clone --filter=blob:none --quiet https://github.com/IDEA-Research/GroundingDINO 'C:\Users\zx\AppData\Local\Temp\pip-req-build-3ase2mik'

  error: RPC failed; curl 92 HTTP/2 stream 3 was not closed cleanly before end of the underlying stream

  error: 5477 bytes of body are still expected

  fetch-pack: unexpected disconnect while reading sideband packet

None
  fatal: early EOF

GroundingDINO install failed. Please submit an issue to https://github.com/IDEA-Research/Grounded-Segment-Anything/issues.
  fatal: index-pack failed

  error: RPC failed; curl 56 OpenSSL SSL_read: Connection was reset, errno 10054

  error: 512 bytes of body are still expected

  fetch-pack: unexpected disconnect while reading sideband packet

  fatal: early EOF

  fatal: index-pack failed

  error: unable to read sha1 file of .asset/GD_GLIGEN.png (85f0be4d0dd4cf520cfd9c4756272a3508cfd66e)

  error: RPC failed; curl 92 HTTP/2 stream 3 was not closed cleanly before end of the underlying stream

  error: 571 bytes of body are still expected

  fetch-pack: unexpected disconnect while reading sideband packet

  fatal: early EOF

  fatal: index-pack failed

  error: unable to read sha1 file of .asset/GD_SD.png (b89fd87a3d6e56d9dc3b34e0d953cfaf792b7ff6)

  fatal: unable to access 'https://ghproxy.com/https://github.com/IDEA-Research/GroundingDINO/': OpenSSL SSL_read: Connection was reset, errno 10054

  error: unable to read sha1 file of .asset/arch.png (30b23f80ac9c45943120144cb1ba15cf3fbbebd0)

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  Auto packing the repository in background for optimum performance.

  See "git help gc" for manual housekeeping.

  fatal: unable to checkout working tree

  warning: Clone succeeded, but checkout failed.

  You can inspect what was checked out with 'git status'

  and retry with 'git restore --source=HEAD :/'

  error: subprocess-exited-with-error

  × git clone --filter=blob:none --quiet https://github.com/IDEA-Research/GroundingDINO 'C:\Users\zx\AppData\Local\Temp\pip-req-build-3ase2mik' did not run successfully.

  │ exit code: 128

  ╰─> See above for output.

  note: This error originates from a subprocess, and is likely not a problem with pip.

error: subprocess-exited-with-error

× git clone --filter=blob:none --quiet https://github.com/IDEA-Research/GroundingDINO 'C:\Users\zx\AppData\Local\Temp\pip-req-build-3ase2mik' did not run successfully.

│ exit code: 128

╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

**Comment by YanXiaoAI** [comment]
same issue with you, and not solved yet~

**Comment by HassanBinHaroon** [comment]
@guytakeiteasy @YanXiaoAI Try this easy and quick inference on Google Colab. [GroundingDINO-Inference](https://github.com/HassanBinHaroon/GroundingDINO-Inference)

**Comment by cloudlakecho** [comment]
I had the same error on running `pip install -e .` . I solved by using this runtime environment:
&nbsp;&nbsp;&nbsp; Ubuntu 18.04
&nbsp;&nbsp;&nbsp; Python: 3.8.0 
&nbsp;&nbsp;&nbsp; CUDA Toolkit: 11.8 

Additionally, after running the command line, I had `import lzma` error at the executing an example code, "demo/inference_on_a_image.py". 

I solved that error with [these steps](https://stackoverflow.com/a/72828943/5595995):
```
sudo apt-get install lzma
sudo apt-get install liblzma-dev
sudo apt-get install libbz2-dev

sudo cp /usr/lib/python3.8/lib-dynload/_bz2.cpython-38-x86_64-linux-gnu.so /usr/local/lib/python3.8/
sudo cp /usr/lib/python3.8/lib-dynload/_lzma.cpython-38-x86_64-linux-gnu.so /usr/local/lib/python3.8/
```

**Comment by HowieHepburn** [comment]
邮件已收到

**Comment by Greg7000** [comment]
> > @guytakeiteasy @YanXiaoAI Try this easy and quick inference on Google Colab. [GroundingDINO-Inference](https://github.com/HassanBinHaroon/GroundingDINO-Inference)
> 
> thanks,i solved it,detail refer to url: [IDEA-Research/Grounded-Segment-Anything#208](https://github.com/IDEA-Research/Grounded-Segment-Anything/issues/208)

I am having the problem, did not find the solution in your link. Can you summarize what you did?

**Comment by smaragben** [comment]
My problem was that I hadn't properly setup CUDA_HOME, LD_LIBRARY_PATH, and my cuda version was not in the path. Hope it helps!
